back/src/modules/user/utils/email.service.ts

```diff
@@ -44,7 +50,15 @@ export class EmailService {
         <a href="http://localhost:${process.env.PORT}/user/reset/${this.token}">Reset Password</a>
         <p>If you did not request this, please ignore this email or contact our support team.</p>
         <p>Thank you,<br>The Support Team</p>`
-        : `<h2>Welcome, ${this.recieverName}. You have successfully created an email</h2><p>You are in the right place. To achieve all benefit from our platform, please verify your account</p><a href=http://localhost:${process.env.PORT}/user/verify/${this.token}>Verify</a>`;
+        : `<h2>Welcome, ${
```

Comment:
It is a very nice design, but I asked ChatGPT for a better design, which uses a design pattern. Here is the code.

```js 
// Factory for creating different types of emails
class EmailFactory {
  static createEmail(type, params) {
    switch (type) {
      case 'passwordReset':
        return new PasswordResetEmail(params);
      case 'welcome':
        return new WelcomeEmail(params);
      case 'twoFA':
        return new TwoFAEmail(params);
      default:
        throw new Error('Invalid email type');
    }
  }
}

// Base Email Class (Strategy Interface)
class Email {
  constructor({ recieverName, recieverEmail, token, baseUrl }) {
    this.recieverName = recieverName;
    this.recieverEmail = recieverEmail;
    this.token = token;
    this.baseUrl = baseUrl || `http://localhost:${process.env.PORT}`;
  }

  getSubject() {
    throw new Error('Method getSubject() must be implemented');
  }

  getContent() {
    throw new Error('Method getContent() must be implemented');
  }
}

// Password Reset Email
class PasswordResetEmail extends Email {
  getSubject() {
    return "Reset Your Password";
  }

  getContent() {
    return `<h2>Hello, ${this.recieverName}</h2>
            <p>You requested a password reset. Click the link below to reset your password:</p>
            <a href="${this.baseUrl}/user/reset/${this.token}">Reset Password</a>
            <p>If you did not request this, please ignore this email or contact support.</p>
            <p>Thank you,<br>The Support Team</p>`;
  }
}

// Welcome Email
class WelcomeEmail extends Email {
  getSubject() {
    return "Welcome to Massive Shop Market Place";
  }

  getContent() {
    return `<h2>Welcome, ${this.recieverName}</h2>
            <p>To enjoy all the benefits of our platform, please verify your account:</p>
            <a href="${this.baseUrl}/user/verify/${this.token}">Verify</a>`;
  }
}

// Two-Factor Authentication Email
class TwoFAEmail extends Email {
  getSubject() {
    return "Two-Factor Authentication Verification";
  }

  getContent() {
    return `<h2>Secure Your Account</h2>
            <p>Please complete your two-factor authentication by verifying your identity:</p>
            <a href="${this.baseUrl}/user/verify-2fa/${this.token}">Verify 2FA</a>`;
  }
}

// Service Class for Sending Emails
class EmailService {
  constructor(sender) {
    this.sender = sender;
    this.apiKey = process.env.SENDINBLUE_KEY;
    this.apiInstance = new Sib.TransactionalEmailsApi();
  }

  async sendEmail(emailType, params) {
    try {
      const email = EmailFactory.createEmail(emailType, params);

      await this.apiInstance.sendTransacEmail({
        sender: { email: this.sender },
        to: [{ email: params.recieverEmail }],
        subject: email.getSubject(),
        htmlContent: email.getContent(),
        name: "Massive Shop Market Place",
      });
    } catch (err) {
      console.error("Error occurred while sending email:", err);
      throw new Error("Failed to send email.");
    }
  }
}
```